reflect.go

```diff
@@ -968,6 +981,19 @@ func nullableFromJSONSchemaTags(tags []string) bool {
 	return false
 }
 
+func isNullable(k reflect.Kind) bool {
+	switch k {
+	case reflect.Pointer,
+		reflect.UnsafePointer,
+		reflect.Map,
+		reflect.Slice,
+		reflect.Interface:
+		return true
```

Comment:
I do think there should be specific tests for these types. I also wonder if `Interface` and `UnsafePointer` are really necessary, if so, there should be a test to show it ;-)

Reply:
@samlown I've added the tests in be77f3d54585a05a330261a21c6dce7bac4d08bb

I can see the appeal to remove both Interface & UnsafePointer as well, so I can update the code accordingly, if you insist.

I also updated the code to actually handle a more varied cases like `map[string]*int` or `*[]*int` & included that in tests.